File: MB-Chat/observability_py/memory_metrics.py
from .types import ObservabilityEvent


def calculate_memory_metrics(events: list[ObservabilityEvent]) -> dict[str, int]:
    scoped = [event for event in events if event.layer == "memory"]
    return {
        "memory_event_count": len(scoped),
        "recall_count": len([event for event in scoped if "recall" in event.event_type]),
        "write_count": len([event for event in scoped if "remember" in event.event_type]),
        "fallback_count": len([event for event in scoped if "fallback" in event.event_type]),
    }

File: MB-Chat/observability/memory-metrics.ts
import type { ObservabilityEvent } from "./types";

export function calculateMemoryMetrics(events: ObservabilityEvent[]) {
  const scoped = events.filter((event) => event.layer === "memory");
  return {
    memory_event_count: scoped.length,
    recall_count: scoped.filter((event) => event.event_type.includes("recall")).length,
    write_count: scoped.filter((event) => event.event_type.includes("remember")).length,
    fallback_count: scoped.filter((event) => event.event_type.includes("fallback")).length,
  };
}
